test/Common/SupportedLanguagesTestAttribute.cs

```diff
@@ -0,0 +1,103 @@
+    /// list of this class (ex: CSharp, JavaScript)
+    /// </summary>
+    public class SqlInlineDataAttribute : DataAttribute
+    {
+        private readonly Dictionary<string, List<List<object>>> testData = new Dictionary<string, List<List<object>>>();
+        private readonly string _propertyName = "data";
+
+        /// <summary>
+        /// Load supported languages as test parameters data for the test.
+        /// </summary>
+        public SqlInlineDataAttribute()
+        {
+            var langData = new List<List<object>>();
+            foreach (string lang in Enum.GetNames(typeof(SupportedLanguages)))
+            {
+                var listOfValues = new List<object>(1) { lang };
+                langData.Add(listOfValues);
+            }
+            this.testData.Add(this._propertyName, langData);
+
+        }
+        /// <summary>
+        /// Load the test parameters data for all supported languages for inline use of the test.
+        /// </summary>
+        /// <param name="args">The test parameters to insert inline for the test</param>
+        public SqlInlineDataAttribute(params object[] args)
+        {
+            var langData = new List<List<object>>();
+            foreach (string lang in Enum.GetNames(typeof(SupportedLanguages)))
+            {
+                var listOfValues = new List<object>();
+                foreach (object val in args)
+                {
+                    listOfValues.Add(val);
+                }
+                listOfValues.Add(lang);
+                langData.Add(listOfValues);
+            }
+            this.testData.Add(this._propertyName, langData);
+        }
+
+        /// <inheritDoc />
+        public override IEnumerable<object[]> GetData(MethodInfo testMethod)
+        {
+            if (testMethod == null) { throw new ArgumentNullException(nameof(testMethod)); }
+            if (this.testData.Count == 0) { throw new EmptyException(nameof(testMethod)); }
+
+            UnsupportedLanguagesAttribute unsupportedLangAttr = testMethod.GetCustomAttribute<UnsupportedLanguagesAttribute>();
+            if (unsupportedLangAttr != null)
+            {
+                var unsupportedLanguages = new List<string>();
+                unsupportedLanguages.AddRange(unsupportedLangAttr.UnsuppportedLanguages);
+                foreach (string lang in unsupportedLanguages)
+                {
+                    List<List<object>> dataList = this.testData.GetValueOrDefault(this._propertyName);
+                    dataList.RemoveAll(l => l.Contains(lang));
+                    this.testData[this._propertyName] = dataList;
+                }
+            }
+
+            string serializedData = JsonConvert.SerializeObject(this.testData);
+            var allData = JObject.Parse(serializedData);
+            JToken data = allData[this._propertyName];
+            return data.ToObject<List<object[]>>();
```

Comment:
I'm pretty sure you don't need all this JSON parsing stuff. This method returns an IEnumerable of object arrays - so basically it returns an enumerable where every item is the parameters to pass to the test method. 

See https://hovermind.com/xunit/custom-data-attribute.html for examples. 

In our case all we're doing is adding the language to the list of args passed in. Each attribute will have a single set of args passed in, and then we iterate over the list of languages supported (filtering out the unsupported ones) and for each language we return an array that consists of the original args + the language.